base_layer/core/src/consensus/consensus_encoding/hashing.rs

```diff
@@ -140,13 +140,40 @@ impl<D: Digest> Write for WriteHashWrapper<D> {
 mod tests {
     use blake2::Blake2b;
     use digest::consts::U32;
+    use tari_common::configuration::Network;
     use tari_crypto::hash_domain;
     use tari_script::script;
 
     use super::*;
 
     hash_domain!(TestHashDomain, "com.tari.test.test_hash", 0);
 
+    #[test]
+    fn network_yields_distinct_hash() {
+        let label = "test";
+        let input = [1u8; 32];
+
+        assert_eq!(*CURRENT_NETWORK.lock().unwrap(), Network::Esmeralda);
+
+        // Generate a mainnet hash
+        *CURRENT_NETWORK.lock().unwrap() = Network::MainNet;
+        let hash_mainnet = DomainSeparatedConsensusHasher::<TestHashDomain, Blake2b<U32>>::new(label)
+            .chain(&input)
+            .finalize();
```

Comment:
These tests look pretty much like what I wrote, which didn't get pushed because I caught a stomach bug and died Thursday night.

I am surprised this works though. I thought this wouldn't release the mutex lock after update, making it impossible for the Domain hasher to read the change?

Reply:
It definitely hangs if you keep the `Mutex` in scope via a `let mut`. But as written, I assume the compiler drops it after each use.

Reply:
ohhhh a subtle difference. Big consequences. Makes sense 👍🏻

Reply:
I do notice more failing integration tests than usual, but this seems to be the case for other recent PRs as well; likely unrelated?